src/group_theory/finiteness.lean

```diff
@@ -264,6 +268,14 @@ group.fg_iff_monoid.fg.mpr $ @monoid.fg_of_surjective G _ G' _ (group.fg_iff_mon
 instance group.fg_range {G' : Type*} [group G'] [group.fg G] (f : G →* G') : group.fg f.range :=
 group.fg_of_surjective f.range_restrict_surjective
 
+variables (G)
+
+/-- The minimum number of generators of a group. -/
+@[to_additive "The minimum number of generators of an additive group"]
+def group.rank [h : group.fg G]
```

Comment:
Could you please provide a bit of API for this new definition? This should be < 20 lines, copying some of the API for `nat.find`. E.g. `group.rank_le` and `group.rank_spec`.